inspirehep/utils/references.py

```diff
@@ -0,0 +1,79 @@
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2015, 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+
+from invenio_records.api import Record
+from invenio_search import current_search_client as es
+from jinja2 import render_template_to_string
+
+
+class Reference(object):
+    """Class used to output reference format in detailed record"""
+
+    def __init__(self, record):
+        self.record = record
+
+    def references(self):
+        """Reference export for single record in datatables format.
+
+        :returns: list
+            List of lists where every item represents a datatables row.
+            A row consists of [reference_number, reference, num_citations]
+        """
+
+        out = []
+        references = self.record.get('references')
+        if references:
+            refs_to_get_from_es = [
+                ref['recid'] for ref in references if ref.get('recid')
```

Comment:
As mentioned in https://github.com/inspirehep/inspire-next/issues/1127 at the moment the references do not contain recids so this code will not work, I am going to create a separate issue for that.

I have used this PR to create the screencast locally so there is no need to merge immediately.

@nikpap can you check https://github.com/inspirehep/inspire-next/issues/1127 and see if recids are present? And then some unit tests of the generation of references and citations should be added.